src/client/index.ts

```diff
@@ -0,0 +1,41 @@
+import Quagga, { QuaggaJSResultObject } from "@ericblade/quagga2"
+import { submitBarcode } from "./lib/api"
+import debounce from "./utils/debounce"
+
+const debouncedSubmit = debounce(submitBarcode, 500)
```

Comment:
Quagga seems to fire its `onDetected` event a bunch of times, so adding a debounce so it doesn't spam the server.

Reply:
I was wondering how likely Quagga is misinterpret a barcode string and if it might be worth building a set of all the values it's reporting and picking the most likely or something.
Though I've not seen that happen, if its recognized a barcode at all it seems to get the value right
I guess barcodes are probably designed to avoid that happening anyway, with checksums and what not

Reply:
Good shout, but you're right about barcode checksums and stuff. I suspect it's a relatively low rate of errors, this is more useful for stopping 19 calls to the API, which mine did one time. Possibly just my laptop being weird (Firefox on Arch on a 5-year-old ThinkPad), so I'll try it on my MacBook too.